packages/ember-data/lib/serializers/json-api-serializer.js

```diff
@@ -0,0 +1,294 @@
+/**
+  @module ember-data
+*/
+
+import Serializer from "ember-data/system/serializer";
+
+var copy = Ember.copy;
+var get = Ember.get;
+var forEach = Ember.ArrayPolyfills.forEach;
+var map = Ember.ArrayPolyfills.map;
+
+/**
+  @class JSONAPISerializer
+  @namespace DS
+  @extends DS.Serializer
+*/
+export default Serializer.extend({
+
+  keyForAttribute: function(key) {
+    return Ember.String.dasherize(key);
+  },
+
+  keyForRelationship: function(key, kind) {
+    return Ember.String.dasherize(key);
+  },
+
+  normalizeTypeKey: function(typeKey) {
+    return Ember.String.camelize(typeKey);
+  },
```

Comment:
My main concern is I'd like to get the default serialization of the `type` property correct now, rather then have to change it in the future. Ideally, this would not be coupled to globals vs modules versions of an Ember App.